Source/WebCore/platform/mediarecorder/cocoa/AudioSampleBufferCompressor.mm

```diff
@@ -94,23 +95,41 @@
     return true;
 }
 
+Ref<GenericPromise> AudioSampleBufferCompressor::drain()
+{
+    return invokeAsync(m_serialDispatchQueue, [weakThis = ThreadSafeWeakPtr { *this }, this] {
+        RefPtr protectedThis = weakThis.get();
+        if (!protectedThis)
+            return GenericPromise::createAndReject();
+
+        m_isDraining = true;
+        processSampleBuffers();
+        m_isDraining = false;
+        if (m_converter) {
+            auto error = PAL::AudioConverterReset(m_converter);
+            RELEASE_LOG_ERROR_IF(error, MediaStream, "AudioSampleBufferCompressor AudioConverterReset failed %d", error);
```

Comment:
Should we reject if reset failed?

Reply:
the recorder always assume they are resolved; just like flush() and finish() currently.

Reply:
Im going to make both promises be rejected if an error occurred (for drain and finish)

Reply:
we don't report debug error in that class anyway, nor to the MediaRecorder.